doc/user/content/sql/system-catalog/mz_internal.md

```diff
@@ -314,6 +343,36 @@ messages and additional metadata helpful for debugging.
+<!-- statement executed, that the system decided to log. Entries older than -->
+<!-- thirty days may be removed. -->
+
+<!-- The system chooses to log statements randomly; the probability of -->
+<!-- logging an execution is controlled by the -->
+<!-- `statement_logging_sample_rate` session variable. A value of 0 means -->
+<!-- to log nothing; a value of 0.8 means to log approximately 80% of -->
+<!-- statement executions. If `statement_logging_sample_rate` is higher -->
+<!-- than `statement_logging_max_sample_rate` (which is set by Materialize -->
+<!-- and cannot be changed by users), the latter is used instead. -->
+
+<!-- | Field                   | Type                         | Meaning                                                                                                                                                                                                                        | -->
+<!-- |-------------------------|------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------| -->
+<!-- | `id`                    | [`uuid`]                     | The ID of the execution event.                                                                                                                                                                                                 | -->
+<!-- | `prepared_statement_id` | [`uuid`]                     | The ID of the prepared statement being executed. Corresponds to [`mz_prepared_statement_history.id`](#mz_prepared_statement_history).                                                                                          | -->
+<!-- | `sample_rate`           | [`double precision`]         | The sampling rate at the time the execution began.                                                                                                                                                                             | -->
+<!-- | `params`                | [`text list`]                | The values of the prepared statement's parameters.                                                                                                                                                                             | -->
+<!-- | `began_at`              | [`timestamp with time zone`] | The time at which execution began.                                                                                                                                                                                             | -->
+<!-- | `finished_at`           | [`timestamp with time zone`] | The time at which execution ended.                                                                                                                                                                                             | -->
```

Comment:
What do you imagine this being? For non-peek statements, an `ExecuteResponse` is ready and can immediately be returned to a user. For `SELECT`s, though, a future is returned and resolved only when clusterd has returned the vec of rows, but that is handled much deeper in the row sending code. See https://github.com/MaterializeInc/materialize/pull/20532 for more discussion about this metric.

What about `SUBSCRIBE`? This could measure how long it took to register the subscribe with clusterd.

What about `FETCH`, which contains the name of another portal that needs to be polled for results, and which may need to begin execution itself?

Reply:
In the version of statement logging that I have working locally, which I will PR in the next 1-2 days or so, the answers are as follows:

* For statements that finish immediately in the coordinator, we are done when we call `ExecuteContext::retire`.
* For `SELECT` statements, we are done when the row future resolves, or immediately in the case of constants that are resolved in the coordinator without a peek ever being sent to a cluster.
* For `SUBSCRIBE` statements, we are done when the subscribe ends (either because it actually finished, which is possible if e.g. subscribing to a constant value, or because the user cancelled it)
* For `FETCH` statements, we are done whenever the inner execute finishes.

The references in [this doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/reference/adapter/coord_interface.md) to when a statement is considered finished are intended to correspond to how they will be treated as part of statement logging.

Reply:
> For `SUBSCRIBE` statements, we are done when the subscribe ends (either because it actually finished, which is possible if e.g. subscribing to a constant value, or because the user cancelled it).

Are we going to emit sampled `SUBSCRIBE` with `finished_at` set to `NULL` until they are done or wait for them to finish before we log them?

*Edit*: I guess having two kinds of entries for "execution begin" vs "execution end" modeled as separate structs means that we're leaning towards the former. I remember from the design doc that we want to do this internally, what I'm not sure is whether this internal state will be exposed as the current state of the corresponding `mz_internal` table.

Reply:
Yes, the publicly-exposed `finished_at` will be `NULL` while the query is still running.